src/xla/Buffer.jl

```diff
@@ -1,16 +1,30 @@
 # Buffer
 @inline function free_buffer(buffer)
-    sbuffer = buffer.buffer
-    if sbuffer != C_NULL
-        @ccall MLIR.API.mlir_c.PjRtBufferFree(sbuffer::Ptr{Cvoid})::Cvoid
+    if buffer.holded == C_NULL
+        if buffer.buffer != C_NULL
+            @ccall MLIR.API.mlir_c.PjRtBufferFree(buffer.buffer::Ptr{Cvoid})::Cvoid
+        end
+    else
+        @ccall MLIR.API.mlir_c.reactant_release_pjrtbuffer(buffer.holded::Ptr{Cvoid})::Cvoid
     end
 end
 
 mutable struct Buffer
     buffer::Ptr{Cvoid}
+    holded::Ptr{Cvoid}
```

Comment:
we should just make a second function HeldBuffer which just contains the heldbuffer pointer, and then for IFRT we shouldn't need to use the old buffer class

Reply:
mmm okay, but we need this `HeldBuffer` inside `Buffer` to avoid it being destructed when IFRT is using it and `Buffer` is GCed.

how about this?
```suggestion
    holded::Union{Nothing,HeldBuffer}
```